li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -424,86 +482,121 @@ synchronized public Long committedSafeOffset(TopicPartition tp) {
 
   @Override
   synchronized public List<PartitionInfo> partitionsFor(String topic) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    return getOrCreateConsumerForTopic(topic).partitionsFor(topic);
   }
 
   @Override
   synchronized public List<PartitionInfo> partitionsFor(String topic, Duration timeout) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    return getOrCreateConsumerForTopic(topic).partitionsFor(topic, timeout);
   }
 
   @Override
   synchronized public Map<String, List<PartitionInfo>> listTopics() {
```

Comment:
Same here, should we remove the synchronized in the outer listTopics() ?

Reply:
done